src/DataTable/TableContext.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const TableContext = createContext({});
+
+export default function TableData({ children }) {
+  const [tableInstances, setTableInstances] = useState({});
+  const setTableInstance = (tableName, tableInstance) => {
+    if (!tableName) {
+      throw new Error('No table name provided to setTableInstance');
+    }
+    setTableInstances({ ...tableInstances, [tableName]: tableInstance });
+  };
```

Comment:
This whole PR is magic, but this is the one piece I'm lost on. Do you think there's a good use case for having multiple tables inside each other where components would need to be able to render display for their "grandparent" table context?

Reply:
Not exactly, but I could definitely think of ways I might want two different tables to interact together. For instance, we basically have a table within a table here: https://portal.stage.edx.org/laelcooffers/admin/coupons
And I could think of ways in which we might want the state of one table to affect another.
I'm not wedded to this idea, but it does allow you to have multiple tables on one page without multiple providers.